app/models/driver.rb

```diff
@@ -0,0 +1,40 @@
+
+class Driver < ApplicationRecord
+  has_many :trips, dependent: :destroy
+
+  # must provide a name
+  validates :name, presence: {message: "%{value} must be given"}
+
+  # must provide a vin
+  validates :vin, presence: {message: "%{value} must be given"}
+
+  def self.select_driver
+    return Driver.all.sample
+  end
+
+  def average_rating
```

Comment:
There are a few ways that we can use Enumerable methods, or even better [ActiveRecord calculation methods](http://api.rubyonrails.org/v5.1/classes/ActiveRecord/Calculations.html), to simplify the implementation of this method. Here are two options:

Enumerable style:
```ruby
def average_rating
  trips.map(&:rating).compact.sum / trips.length
end
```

ActiveRecord style:
```ruby
def average_rating
  trips.average(:rating)
end
```

The second option is my preferred style because it is both simpler code and more efficient. This version actually moves the calculation of the average into SQL code, so the Postgres database ends up doing the average calculation for us. In that case it is significantly faster when we have a very large amount because it doesn't need to be sent to Rails first.